### PR TITLE
[main] [kie-issues#522] Bump Quarkus version to 2.16.10.Final

### PR DESCRIPTION
**Pull request by kie-ci3**
https://github.com/kiegroup/kie-issues/issues/522

Please review and merge

- https://github.com/kiegroup/kogito-images/pull/1679
- https://github.com/kiegroup/kogito-docs/pull/470
- https://github.com/kiegroup/kogito-runtimes/pull/3187
- https://github.com/kiegroup/kogito-examples/pull/1772

### CODE REVIEW

File: kogito-build/kogito-dependencies-bom/pom.xml [hunk 1]
```diff
@@ -16,10 +16,10 @@
 
   <properties>
     <!-- this version property is used in plugins but also in dependencies too -->
-    <version.io.quarkus>2.16.9.Final</version.io.quarkus>
+    <version.io.quarkus>2.16.10.Final</version.io.quarkus>
     <version.io.quarkus.quarkus-test>${version.io.quarkus}</version.io.quarkus.quarkus-test>
     <version.org.springframework.boot>2.6.6</version.org.springframework.boot>
-    <version.org.apache.kafka>3.4.0</version.org.apache.kafka>
+    <version.org.apache.kafka>3.3.2</version.org.apache.kafka>
```
Comment (radtriste) [inline kogito-build/kogito-dependencies-bom/pom.xml:22]:
```suggestion
    <version.org.apache.kafka>3.4.0</version.org.apache.kafka>
```

File: kogito-build/kogito-dependencies-bom/pom.xml [hunk 2]
```diff
@@ -124,7 +124,7 @@
     <!-- see: https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html#parallel-test-execution-and-single-thread-execution -->
     <version.com.github.stephenc.jcip>1.0-1</version.com.github.stephenc.jcip>
     <version.black.ninia>4.1.1</version.black.ninia>
-    <version.com.google.guava>32.0.1-jre</version.com.google.guava>
+    <version.com.google.guava>31.1-jre</version.com.google.guava>
```
Comment (radtriste) [inline kogito-build/kogito-dependencies-bom/pom.xml:127]:
```suggestion
    <version.com.google.guava>32.0.1-jre</version.com.google.guava>
```

**Review by radtriste — APPROVED** [review]

**Comment by kie-ci3** [comment]
**(tests) - kogito-runtimes job** `#3085` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-T 1C -Dvalidate-formatting -Prun-code-coverage
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-runtimes -u https://github.com/kiegroup/kogito-runtimes/pull/3187 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/3085/display/redirect

**Test results:**
- PASSED: 3367
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/3085/testReport/org.kie.kogito.persistence.kafka/KafkaProcessInstancesIT/testFindByIdReadMode/">org.kie.kogito.persistence.kafka.KafkaProcessInstancesIT.testFindByIdReadMode</a></summary>
<br/>expected: 5<br/> but was: 0
</details>

**Comment by radtriste** [comment]
jenkins rerun kogito-runtimes tests

**Pull request merged**